pkg/go/transformer/dsltojson.go

```diff
@@ -107,24 +122,30 @@ func (l *openFgaDslListener) ExitRelationDeclaration(ctx *parser.RelationDeclara
 		}
 	}
 
-	directlyRelatedUserTypes := l.currentRelation.TypeInfo.GetDirectlyRelatedUserTypes()
+	if relationDef != nil {
+		if l.currentTypeDef.Relations[relationName] != nil {
+			ctx.GetParser().NotifyErrorListeners(fmt.Sprintf("%s is already defined in %s", relationName, l.currentTypeDef.Type), ctx.GetParser().GetCurrentToken(), nil)
+		}
 
-	l.currentTypeDef.Metadata.Relations[relationName] = &pb.RelationMetadata{DirectlyRelatedUserTypes: directlyRelatedUserTypes}
+		l.currentTypeDef.Relations[relationName] = relationDef
```

Comment:
Need to check whether l.currentTypeDef is not nil?

Reply:
Already initialized here: https://github.com/openfga/language/blob/50b003d312da3fdb52812a562ea22292bc186e6e/pkg/go/transformer/dsltojson.go#L48